public/app/features/plugins/admin/state/hooks.ts

```diff
@@ -133,6 +133,12 @@ export const useFetchDetails = (id: string) => {
   }, [plugin]); // eslint-disable-line
 };
 
+export const useFetchDetailsLazy = () => {
+  const dispatch = useDispatch();
+
+  return (id: string) => dispatch(fetchDetails(id));
+}
+
```

Comment:
Although this should work, maybe what we could do instead is to add a way to force-fetch with the existing `useFetchDetails()` hook - in that case we would only have a single way for fetching plugin details using hooks.

E.g.:
```typescript
export const useFetchDetails = (id: string, { force = false } = {}) => {
  const dispatch = useDispatch();
  const plugin = useSelector((state) => selectById(state, id));
  const isNotFetching = !useSelector(selectIsRequestPending(fetchDetails.typePrefix));
  const shouldFetch = force || (isNotFetching && plugin && !plugin.details);

  useEffect(() => {
    shouldFetch && dispatch(fetchDetails(id));
  }, [plugin]); // eslint-disable-line
};
```

Reply:
@leventebalogh  tks for your review.

I tried a similar approach before, removing the `!plugin.details` from `shouldFetch` caused a infinite loop. The main problem is that `fetchDetails` is inside `useEffect` in this existing hook. That's why I went for the lazy fetch.

The suggested approach will cause the same.
Just to check if I'm not doing anything wrong: I changed the `useFetchDetails`, as you suggested, and then added the hook in `InstallControlsButton` component right before the first `useEffect` that is already there.

Reply:
Yeah, you are actually right - I missed that `useFetchDetails()` is not returning with a function. 
Let's leave it as it is 👍